### PR TITLE
feat: support generate maven dependency tree

### PR DESCRIPTION
**Pull request by aiwantaozi**

### CODE REVIEW

File: syft/pkg/cataloger/java/scaffolding_parser.go [hunk 1]
```diff
@@ -145,40 +152,48 @@ func (p *scaffoldingParser) parseMaven(ctx context.Context) (pkgs []*pkg.Package
...
-			continue
+
+		var toPkg *pkg.Package
+		var relation *artifact.Relationship
+
+		if strings.Contains(line, "->") {
+			// type: dependency
+			// line example: "org.apache.logging.log4j:log4j-core:jar:2.18.0:compile" -> "org.apache.logging.log4j:log4j-api:jar:2.18.0:compile" ;
+			toPkg, relation, err = p.parseMavenRelationLine(line, mainPkgPomProject)
+		} else if strings.Contains(line, "digraph") {
```
Comment (gitlawr) [inline syft/pkg/cataloger/java/scaffolding_parser.go]: What if it's a part of the package name?

File: syft/pkg/java_metadata.go [hunk 2]
```diff
@@ -21,13 +21,14 @@ var jenkinsPluginPomPropertiesGroupIDs = []string{
...
+	VirtualPath          string         `json:"virtualPath" cyclonedx:"virtualPath"` // we need to include the virtual path in cyclonedx documents to prevent deduplication of jars within jars
+	Manifest             *JavaManifest  `mapstructure:"Manifest" json:"manifest,omitempty"`
+	PomProperties        *PomProperties `mapstructure:"PomProperties" json:"pomProperties,omitempty" cyclonedx:"-"`
+	PomProject           *PomProject    `mapstructure:"PomProject" json:"pomProject,omitempty"`
+	ArchiveDigests       []file.Digest  `hash:"ignore" json:"digest,omitempty"`
+	PURL                 string         `hash:"ignore" json:"-"` // pURLs and CPEs are ignored for package IDs
+	Parent               *Package       `hash:"ignore" json:"-"` // note: the parent cannot be included in the minimal definition of uniqueness since this field is not reproducible in an encode-decode cycle (is lossy).
+	OverwriteRootPackage bool           `json:"overwriteRootPackage,omitempty" cyclonedx:"overwriteRootPackage"`
```
Comment (gitlawr) [inline syft/pkg/java_metadata.go]: Can you add a comment to explain

**Review by thxCode — APPROVED** [review]
LGTM

**Review by gitlawr — APPROVED** [review]
LGTM

**Pull request merged**